portal/src/components/UU/UU.tsx

```diff
@@ -0,0 +1,278 @@
+            return node.frontmatter.tags.some((t) =>
+                checkedTags.find(([filterTag, { checked }]) => filterTag === t && checked),
+            );
+        },
+        [tagFilter],
+    );
+
+    const filterBySearch = useCallback(
+        (node: MDXNode) => {
+            const bodyMatch = node.body.includes(search);
+            const titleMatch = node.frontmatter.title.toLowerCase().includes(search.toLowerCase());
+            const wcagMatch = node.frontmatter.wcagRules ? node.frontmatter.wcagRules.includes(search) : true;
+
+            return bodyMatch || titleMatch || wcagMatch;
+        },
+        [search],
+    );
+
+    // check if a filter is enabled
+    const hasFilter = useMemo(() => {
+        const hasSearchFilter = !!search;
+        const hasTagsFilter = !!Object.entries(tagFilter).filter(([, { checked }]) => checked).length;
+
+        return hasSearchFilter || hasTagsFilter;
+    }, [search, tagFilter]);
+
+    const filteredNodes = useMemo(() => {
+        if (!data) {
+            return [];
+        }
+
+        if (!hasFilter) {
+            return data.allMdx.nodes;
+        }
+
+        return data.allMdx.nodes.filter(filterByTag).filter(filterBySearch);
+    }, [data, hasFilter, filterBySearch, filterByTag]);
+
+    return (
+        <section className="jkl-portal__uu">
+            <section className="uu__section--search">
+                <FieldGroup legend="Hva inneholder løsningen din?">
+                    {Object.entries(tagFilter).map(([tag, { checked }]) => (
+                        <Checkbox name="uu-area-tag" value={tag} key={tag} checked={checked} onChange={onTagChange}>
+                            {tagMap[tag as Tag]}
+                        </Checkbox>
+                    ))}
+                </FieldGroup>
+
+                <PrimaryButton onClick={scrollToResults}>Vis resultat</PrimaryButton>
+            </section>
+
+            <section className="uu__section--search-results" ref={resultWrapperRef}>
+                <TextInput
+                    variant="small"
+                    width="316px"
+                    label="Søk"
+                    placeholder="Søk i artikler"
+                    value={search}
+                    onChange={onSearchChange}
+                    className="uu__filter-search"
+                    action={hasFilter ? { icon: "clear", label: "Fjern søk", onClick: handleClear } : undefined}
+                    aria-label="Søk i artikler"
+                />
+
+                {hasFilter && <h2>Resultat</h2>}
+
+                <ul className="uu__filter-result-list">
+                    {hasFilter &&
+                        filteredNodes.map((node) => (
+                            <motion.li
+                                key={node.id}
+                                initial={{ y: prefersReducedMotion ? 0 : -40, opacity: 0 }}
+                                animate={{
+                                    y: prefersReducedMotion ? 0 : 0,
```

Comment:
Denne ser ganske meningsløs ut

```suggestion
                                    y: 0,
```

Reply:
🤦